File: 2015/Day_06/Python/test_part_1.py
from . import part_1


def test_generate_coordinates():
    assert len(part_1.generate_coordinates((0, 0), (2, 2))) == 9
    assert part_1.generate_coordinates((0, 0), (2, 2)) == [(0, 0), (0, 1), (0, 2), (1, 0), (1, 1), (1, 2), (2, 0),
                                                           (2, 1), (2, 2)]


def test_parse_instructions():
    assert part_1.parse_instructions('turn on 887,9 through 959,629') == ("turn on", (887, 9), (959, 629))
    assert part_1.parse_instructions('toggle 720,196 through 897,994') == ("toggle", (720, 196), (897, 994))


def test_run_light_show():
    instructions = ['turn on 0,0 through 2,2', 'toggle 1,1 through 2,2']
    assert part_1.run_the_light_show(instructions) == 5
    instructions = ['toggle 1,1 through 2,2']
    assert part_1.run_the_light_show(instructions) == 4
    instructions = ['turn on 0,0 through 2,2', 'turn off 0,0 through 2,2']
    assert part_1.run_the_light_show(instructions) == 0
    instructions = ['turn on 0,0 through 999,999']
    assert part_1.run_the_light_show(instructions) == 1000000
    instructions = ['toggle 0,0 through 2,2']
    assert part_1.run_the_light_show(instructions) == 9

File: 2015/Day_06/Ruby/test_part_1.rb
require_relative "part_1"
require 'test/unit'

class TestLightCoordinates < Test::Unit::TestCase
    
    def test_generate_coordinates
        coordinates = generate_coordinates([0,0], [2,2])
        assert_equal(9, coordinates.length)
    end
    
    def test_get_coordinates_from_text
        the_matches = get_coordinates_from_text("turn on 0,0 through 999,999")
        starting_x  = the_matches[0][0].to_i
        starting_y  = the_matches[0][1].to_i
        ending_x = the_matches[1][0].to_i
        ending_y = the_matches[1][1].to_i
        assert_equal(0, starting_x)
        assert_equal(0, starting_y)
        assert_equal(999, ending_x)
        assert_equal(999, ending_y)
    end
end
